### ISSUE TITLE
No `(throw :finish t)` in some parsers.

### ISSUE THREAD

**Issue by c1-g** [yantar92/org-capture-ref#3]
I've set this up with qutebrowser userscript copied from the README file, which you can see [here](https://gitlab.com/c1-g/dotfiles/-/blob/fbsd/.config/qutebrowser/userscripts/org-capture-ref). I modified it to use the `z` and `Z` templates.
When I called the userscript all I get were the message "Capturing using generic parser" and "Bookmark not saved", without the capture process. I tried calling the userscript on this repo page just like in your example and no capture process still. Then I check the value of my `org-capture-ref--bibtex-alist` and it was
``` emacs-lisp
((:year . "unused")
 (:title . "yantar92/org-capture-ref: Extract metadata/bibtex info from websites for org-capture")
 (:author . "yantar92")
 (:doi . "unused")
 (:howpublished . "Github")
 (:url . "https://github.com/yantar92/org-capture-ref")
 (:created . "[2021-12-23 Thu 08:40]")
 (:urldate . "23 December 2021")
 (:type . "misc"))
```
So the parsers seem to be doing their job but somehow no capture buffer is being raised.

Adding `(throw :finish t)` to the end of `org-capture-ref-get-bibtex-github-repo` seems to solve the issue.

And I see that `org-capture-ref-get-bibtex-srht-repo`, `org-capture-ref-get-bibtex-amazon`, `org-capture-ref-get-bibtex-lesswrong` and more parsers are missing the `(throw :finish t)` as well.

Is this a deliberate design choice or a mistake?

**Comment by yantar92** [comment]
> When I called the userscript all I get were the message "Capturing using generic parser" and "Bookmark not saved"

This usually indicates some error thrown in the middle of capture process. I should really improve the error reporting in this area.

Can you `debug-on-entry` on `org-capture-ref-process-capture` while capturing the problematic website and check which capture processing step throws an error (the suspect is `org-capture-ref-parse-generic`)?

> And I see that org-capture-ref-get-bibtex-srht-repo, org-capture-ref-get-bibtex-amazon, org-capture-ref-get-bibtex-lesswrong and more parsers are missing the (throw :finish t) as well.
> 
> Is this a deliberate design choice or a mistake?

For amazon and github, it is deliberate. They are hard to parse because the websites change page layout from time to time. Not throwing finish allows to fallback to generic parser for non-standard pages.

In general, `(throw :finish t)` is intended to use with parsers that know for sure that they successfully parse the website and all the necessary fields within that specific parser. It should normally be optional.

**Comment by c1-g** [comment]
Found the culprit! `delete-if` in `org-capture-ref-query-dom`
is causing this error:
`Symbol’s function definition is void: delete-if`

The function `delete-if` is a function from `cl-lib` but it has since been renamed to `cl-delete-if`.

A solution is either changing the name of the function to `cl-delete-if` and add `(require cl-lib)` or simply replace it with `delete` equivalent.

Works great now, thanks!
